modules/core/src/directx.cpp

```diff
@@ -41,9 +41,19 @@
 
 #include "precomp.hpp"
 
+#ifdef HAVE_OPENCL
+
+#if defined __APPLE__
+#include <OpenCL/opencl.h>
+#else
+#include <CL/opencl.h>
+#endif
+#endif
```

Comment:
Mac should not use anything from this file. DirectX is Windows only feature.
Anyway OpenCV code should not use OpenCL files directly: https://github.com/Itseez/opencv/blob/6b8e13d01725b8dbba97c365f29afc44bafe8cd2/modules/core/include/opencv2/core/opencl/runtime/opencl_core.hpp